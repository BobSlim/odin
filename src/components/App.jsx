import { Icon } from "@iconify/react"
import { Outlet, Link } from "react-router-dom"

const NavBar = () =>
    <nav>
        <Link to="/"><button>Home</button></Link>
        <Link to="/careers"><button>Careers</button></Link>
        <Link to="/game"><button>Game</button></Link>
        <Link to="/shop"><button>Shop</button></Link>
        <CartButton />
    </nav>

const CartButton = ({ items = 0 }) =>
    <button>
        <Icon icon="mdi:cart"></Icon>
        {items}
    </button>

export const App = () => {
    return (
        <>
            <NavBar />
            <Outlet />
        </>
    )
}