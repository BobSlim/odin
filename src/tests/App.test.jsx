import { App } from '@/components/App.jsx'
import { render, screen } from '@testing-library/react';


describe('App', () => {
    it('renders', () => {
        render(<App />);
        screen.debug();

        // check if App components renders headline
    });
});