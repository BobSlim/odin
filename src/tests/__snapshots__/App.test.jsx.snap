// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <button>
        Home
      </button>
    </a>
    <a
      href="/careers"
    >
      <button>
        Careers
      </button>
    </a>
    <a
      href="/game"
    >
      <button>
        Game
      </button>
    </a>
    <a
      href="/shop"
    >
      <button>
        Shop
      </button>
    </a>
    <button>
      <span />
      0
    </button>
  </nav>
</div>
`;
